aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -431,29 +434,38 @@ module aptos_framework::stake {
         aptos_framework: &signer,
         validators: &vector<address>,
     ) acquires ValidatorSet {
+        assert_reconfig_not_in_progress();
         system_addresses::assert_aptos_framework(aptos_framework);
-
         let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
         let active_validators = &mut validator_set.active_validators;
         let pending_inactive = &mut validator_set.pending_inactive;
-        let len = vector::length(validators);
+        spec {
+            update ghost_active_num = len(active_validators);
```

Comment:
what does ghost mean?

Reply:
let's see what @rahxephon89  says

Reply:
This is used to prove that len(active_validators) + len(pending_inactive) is a invariant (does not change) during execution of remove_validators